test/unit/host_test.rb

```diff
@@ -1519,16 +1519,24 @@ def teardown
       end
     end
 
-    test "built should send host built mail" do
+    test "built should send notification when installed" do
       ActionMailer::Base.deliveries = []
       User.current.mail_notifications << MailNotification[:host_built]
       host = FactoryGirl.create(:host, :managed, :owner => User.current)
-      host.built
+      host.built(true)
       email = ActionMailer::Base.deliveries.detect { |mail| mail.subject =~ /Host #{host} is built/ }
       assert email
       assert_match /Your host has finished/, email.body.encoded
     end
 
+    test "built should not send notifiation if not installed" do
+      ActionMailer::Base.deliveries = []
+      User.current.mail_notifications << MailNotification[:host_built]
+      host = FactoryGirl.create(:host, :managed, :owner => User.current)
```

Comment:
Can you move these setup lines this method and the previous one to a `setup` block, in a `context` 'built notifications'?